base/dict.jl

```diff
@@ -350,9 +386,13 @@ ht_keyindex2!(h::Dict, key) = ht_keyindex2_shorthash!(h, key)[1]
 
     sz = length(h.keys)
     # Rehash now if necessary
-    if h.ndel >= ((3*sz)>>2) || h.count*3 > sz*2
-        # > 3/4 deleted or > 2/3 full
-        rehash!(h, h.count > 64000 ? h.count*2 : h.count*4)
+    if (h.count + h.ndel)*3 > sz*2
```

Comment:
I'd want to benchmark, but IMO since we have the slots to give us fast comparisons, I think we should probably allow for 3/4ths full.

Reply:
This is still 3/4 full, but including deleted slots because all non-zero slots influences average length of linear probing. Yes, only proper benchmarking may justify this change. Also, this is kind of orthogonal to the main idea of this PR.